model/host/host.go

```diff
@@ -1194,7 +1222,18 @@ func StaleRunningTaskIDs(staleness time.Duration) ([]task.Task, error) {
 		}},
 		{"$match": bson.M{
 			"$expr": bson.M{
-				"$eq": []string{"$_id", bsonutil.GetDottedKeyName("$host", RunningTaskKey)},
+				"$and": []bson.M{
+					{"$eq": []string{"$_id", bsonutil.GetDottedKeyName("$host", RunningTaskKey)}},
+					{"$or": []bson.M{ // this expression checks that the host is not currently running teardown_group of a different task
+						{"$not": []bson.M{{"$ifNull": []interface{}{bsonutil.GetDottedKeyName("$host", RunningTeardownForTaskKey), nil}}}},
```

Comment:
Can you really use a $not with an $ifNull? How does this behave?

Reply:
It behaves similarly as in javascript - it'll invert the value in a truthy/not truthy way - https://docs.mongodb.com/manual/reference/operator/aggregation/not/#behavior. This is me trying to do a $exists in an expression